### ISSUE TITLE
sql: TEXT not recognized as alias for STRING in array types

### ISSUE THREAD

**Issue by espadrine** [cockroachdb/cockroach#21850]
Config information:
```
I180127 15:52:05.996315 1 util/log/clog.go:1031  [config] file created at: 2018/01/27 15:52:05
I180127 15:52:05.996315 1 util/log/clog.go:1031  [config] running on machine: tyl-desktop
I180127 15:52:05.996315 1 util/log/clog.go:1031  [config] binary: CockroachDB CCL v1.1.4 (linux amd64, built 2018/01/08 17:32:42, go1.8.3)
I180127 15:52:05.996315 1 util/log/clog.go:1031  [config] arguments: [cockroach start --host=localhost --cache=128MiB --max-sql-memory=128MiB --certs-dir=certs]
```

Reproducible: always.

Setup:
```
root@:26257/dev> CREATE TABLE issues (id TEXT, sessions TEXT[]);
+----------+----------+------+---------+---------+
|  Field   |   Type   | Null | Default | Indices |
+----------+----------+------+---------+---------+
| id       | STRING   | true | NULL    | {}      |
| sessions | STRING[] | true | NULL    | {}      |
+----------+----------+------+---------+---------+
```
**Problem**: The TEXT type is equivalent to STRING, but this query fails:
```
root@:26257/dev> SELECT sessions::TEXT[] FROM issues;
pq: invalid cast: string[] -> TEXT[]
```
Yet the following query does not fail:
```
root@:26257/dev> SELECT id::text FROM issues;
+----------+
| id::TEXT |
+----------+
+----------+
```
I expected to see the following:
```
root@:26257/dev> SELECT sessions::TEXT[] FROM issues;
+------------------+
| sessions::TEXT[] |
+------------------+
+------------------+
```
(This issue is made sadder given that CockroachDB automatically converts TEXT to STRING when creating the database.)

Parity assessment: the expected behavior works as intended on PostgreSQL.

```
psql (9.5.10)
Type "help" for help.

postgres=# create table issues (id text, sessions text[]);
CREATE TABLE
postgres=# select sessions::text[] from issues;
 sessions 
----------
(0 rows)
```

**Comment by knz** [comment]
Hello @espadrine! Thanks for reporting this issue. Looking into it.

**Comment by knz** [comment]
Hi @espadrine!
We have fixed this bug already in CockroachDB 2.0, due to be released soon. Are you willing and able to upgrade? Depending on how critical this bug is in your deployment we may consider a backport into 1.1.

**Comment by knz** [comment]
> 2.0 is due in April or May, correct? 

Yes correct!
